src/components/victory-bar/victory-bar.js

```diff
@@ -141,17 +141,19 @@ class VictoryBar extends React.Component {
 
   renderData(props) {
     const { dataComponent, labelComponent, groupComponent } = props;
-    const dataComponents = [];
-    const labelComponents = [];
-    for (let index = 0, len = this.dataKeys.length; index < len; index++) {
+
+    const dataComponents = this.dataKeys.map((_dataKey, index) => {
       const dataProps = this.getComponentProps(dataComponent, "data", index);
-      dataComponents[index] = React.cloneElement(dataComponent, dataProps);
+      return React.cloneElement(dataComponent, dataProps);
+    });
 
+    const labelComponents = compact(this.dataKeys.map((_dataKey, index) => {
```

Comment:
GLOBAL NOTE: If we don't need to take a dependency on `compact`, then we can simply do this natively with:

```js
const dataComponents = this.dataKeys
  .map((_dataKey, index) => { /* RETURN EMPTY OR VALUE */ })
  .filter(Boolean);
```